Source/Data/Models/Device.swift

```diff
@@ -1,36 +1,74 @@
-    
-    public enum PushType: Int {
+public struct Device: Codable, Equatable {
+
+    public enum PushType: Int, Codable, Equatable {
         case customWebhook = 4
         case customWebhookBehindOAuth = 5
         case fcm = 6
     }
     
+    private enum DeviceType: Int, Codable, Equatable {
+        case android = 1
+        case iOS = 2
+        case web = 3
+        case generic = 4 // Custom build
+    }
+
     var pushToken: String?
     var pushType: PushType?
     var pushEnabled: Bool?
     var userAdditionalInformation: [String: String]?
-    var type: Int = 2 // iOS
-    var version: String = kParleyVersion
+    private var type: DeviceType
+    var version: String
     var referrer: String?
+
+    init(
+        pushToken: String?,
+        pushType: PushType?,
+        pushEnabled: Bool?,
+        userAdditionalInformation: [String : String]?,
+        referrer: String?
+    ) {
+        self.pushToken = pushToken
+        self.pushType = pushType
+        self.pushEnabled = pushEnabled
+        self.userAdditionalInformation = userAdditionalInformation
+        type = DeviceType.iOS
+        version = kParleyVersion
+        self.referrer = referrer
+    }
     
-    init() {
-        //
+    enum CodingKeys: CodingKey {
+        case pushToken
+        case pushType
+        case pushEnabled
+        case userAdditionalInformation
+        case type
+        case version
+        case referrer
     }
     
-    required public init?(map: Map) {
-        //
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.pushToken = try container.decodeIfPresent(String.self, forKey: .pushToken)
+        self.pushType = try container.decodeIfPresent(Device.PushType.self, forKey: .pushType)
+        self.pushEnabled = try container.decodeIfPresent(Bool.self, forKey: .pushEnabled)
+        self.userAdditionalInformation = try container.decodeIfPresent([String : String].self, forKey: .userAdditionalInformation)
+        self.type = try container.decodeIfPresent(DeviceType.self, forKey: .type)  ?? DeviceType.iOS
+        self.version = try container.decodeIfPresent(String.self, forKey: .version) ?? kParleyVersion
+        self.referrer = try container.decodeIfPresent(String.self, forKey: .referrer)
     }
     
-    public func mapping(map: Map) {
-        pushToken <- map["pushToken"]
-        pushType <- (map["pushType"], EnumTransform<PushType>())
-        pushEnabled <- map["pushEnabled"]
-        userAdditionalInformation <- map["userAdditionalInformation"]
-        type <- map["type"]
-        version <- map["version"]
-        referrer <- map["referrer"]
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encodeIfPresent(self.pushToken, forKey: .pushToken)
+        try container.encodeIfPresent(self.pushType, forKey: .pushType)
```

Comment:
encode as deviceType.rawValue (see issue with messages encoding)

Reply:
Changed in: [abdc62c](https://github.com/parley-messaging/ios-library/pull/53/commits/abdc62c612f16978356da27b6a4bdb894a943535)